05.wc/wc.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def exec
+  opt = OptionParser.new
+  params = { number_of_lines: false, number_of_words: false, bytesize: false }
+  opt.on('-l') { |v| params[:number_of_lines] = v }
+  opt.on('-w') { |v| params[:number_of_words] = v }
+  opt.on('-c') { |v| params[:bytesize] = v }
+  opt.parse!(ARGV)
+  params.transform_values!(&:!) unless params.values.any?
+
+  ARGV.empty? ? wc_stdin(params) : wc_files(params)
+end
+
+def wc_stdin(params)
+  lines = ARGF.to_a
+  stat = build_stat(lines)
+  print_format(params, stat)
+end
+
+def wc_files(params, paths: ARGV)
+  path_stats =
+    paths.map do |path|
+      lines = File.read(path).lines
+      build_stat(lines, path:)
+    end
+
+  path_stats << build_total(path_stats) if path_stats.size > 1
+  path_stats.each { |stat| print_format(params, stat) }
+end
+
+def build_stat(lines, path: nil)
+  number_of_lines = lines.size
+  words = lines.join.split(/\s+/)
+  number_of_words = words.size
+  bytesize = lines.map(&:bytesize).sum
+
+  stat = {}
+  stat[:number_of_lines] = number_of_lines
```

Comment:
```suggestion
  {
    number_of_lines: number_of_lines,
```

こんな感じでハッシュリテラルとして直接書いてしまうのもありかもしれません。

Reply:
おっしゃる通り、ハッシュリテラルとして直接書いたほうが、よりスッキリしてわかりやすいですね！
こちら変更しました。